akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -155,6 +155,7 @@ object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
     case _: ActorInitializationException ⇒ Stop
     case _: ActorKilledException         ⇒ Stop
     case _: DeathPactException           ⇒ Stop
+    case m: IncompatibleClassChangeError ⇒ detectAkkaBinaryIncompatibilityWarningAndStop(m)
```

Comment:
This is a `LinkageError` so I don't think it will end up here. not caught by `NonFatal`. Instead, move the check to  ActorSystem